### PR TITLE
Feat: squash dict hint + unit tests

### LINKED ISSUE #419: BuiltinHint: implement squash dict utils
We need to implement squash dict utils for our builtin hint processor.

Rust reference: https://github.com/lambdaclass/cairo-vm/blob/main/vm/src/hint_processor/builtin_hint_processor/squash_dict_utils.rs

### PR DESCRIPTION
**Pull request by StringNick**
closes #419

### CODE REVIEW

File: src/math/fields/fields.zig [hunk 1]
```diff
@@ -686,6 +686,26 @@ pub fn Field(comptime F: type, comptime modulo: u256) type {
...
+                @intCast(std.math.maxInt(usize)),
+            )) {
+                return error.ValueTooLarge;
+            }
+
+            // Otherwise, it's safe to cast
+            return @as(
+                usize,
+                @intCast(asU256),
+            );
```
Comment (tcoratger) [inline src/math/fields/fields.zig:706]:
```suggestion
            return @intCast(asU256);
```

File: src/hint_processor/squash_dict_utils.zig [hunk 2]
```diff
@@ -0,0 +1,1428 @@
...
+) !void {
+    //Check that access_indices and key are in scope
+    const key = try exec_scopes.getValue(Felt252, "key");
+    const range_check_ptr = try hint_utils.getPtrFromVarName("range_check_ptr", vm, ids_data, ap_tracking);
+    const access_indices = try getAccessIndices(exec_scopes);
+    //Get current_indices from access_indices
+    var current_access_indices = try (access_indices.get(key) orelse return HintError.NoKeyInAccessIndices).clone();
+    errdefer current_access_indices.deinit();
+
+    std.sort.block(Felt252, current_access_indices.items[0..], {}, cmpByValue);
```
Comment (tcoratger) [inline src/hint_processor/squash_dict_utils.zig:68]:
```suggestion
    std.sort.block(Felt252, current_access_indices.items, {}, cmpByValue);
```

File: src/vm/types/execution_scopes.zig [hunk 3]
```diff
@@ -340,34 +349,51 @@ pub const ExecutionScopes = struct {
...
         const val = try self.getRef(name);
 
-        if (std.meta.activeTag(val.*) == T) return &@field(val.*, @tagName(T));
+        inline for (@typeInfo(HintType).Union.fields) |field| {
+            if (field.type == T) {
+                if (std.mem.eql(u8, @tagName(val.*), field.name)) {
+                    return &@field(val.*, field.name);
+                }
+
+                return HintError.VariableNotInScopeError;
```
Comment (tcoratger) [inline src/vm/types/execution_scopes.zig:392]:
```suggestion
              return  if (std.mem.eql(u8, @tagName(val.*), field.name))
                    &@field(val.*, field.name) else
            HintError.VariableNotInScopeError;
```

File: src/hint_processor/squash_dict_utils.zig [hunk 4]
```diff
@@ -0,0 +1,1428 @@
...
+    if (!n_used_accesses.equal(Felt252.fromInt(usize, access_indices_at_key.items.len)))
+        return HintError.NumUsedAccessesAssertFail;
+}
+
+// Implements Hint: assert len(keys) == 0
+pub fn squashDictInnerAssertLenKeys(
+    exec_scopes: *ExecutionScopes,
+) !void {
+    //Check that current_access_indices is in scope
+    const keys = try exec_scopes.getValue(std.ArrayList(Felt252), "keys");
```
Comment (tcoratger) [inline src/hint_processor/squash_dict_utils.zig:197]:
```suggestion
    var keys = try exec_scopes.getValue(std.ArrayList(Felt252), "keys");
defer keys.deinit();
```

Comment (StringNick) [reply src/hint_processor/squash_dict_utils.zig:197]: we are not cloning here, so i dont think that we need to deallocate here

**Review by lana-shanghai — APPROVED** [review]
lgtm

**Pull request merged**